copy/copy.go

```diff
@@ -126,6 +127,16 @@ type Options struct {
 	// Download layer contents with "nondistributable" media types ("foreign" layers) and translate the layer media type
 	// to not indicate "nondistributable".
 	DownloadForeignLayers bool
+
+	// Contains slice of OptionCompressionVariant, where c/image will ensure that selected
+	// images in the manifest list will exist with compression specificed in OptionCompressionVariant
+	// while performing copy.
+	EnsureCompressionVariantsExist []OptionCompressionVariant
+}
+
+type OptionCompressionVariant struct {
+	Algorithm *compression.Algorithm
+	Level     int
```

Comment:
Also document something like `// Only used when we are creating a new image instance using the specified algorithm, not when the image already contains such an instance`